catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -142,4 +209,43 @@ public PropertiesMetadata filesetPropertiesMetadata() throws UnsupportedOperatio
   public PropertiesMetadata tablePropertiesMetadata() throws UnsupportedOperationException {
     throw new UnsupportedOperationException("Kafka catalog does not support table operations");
   }
+
+  private void createDefaultSchema() {
+    // If the default schema already exists, do nothing
+    try {
+      if (store.exists(defaultSchemaIdent, Entity.EntityType.SCHEMA)) {
+        return;
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to check if schema " + defaultSchemaIdent + " exists", e);
+    }
+
+    // Create the default schema
+    long uid = idGenerator.nextId();
+    ImmutableMap<String, String> properties =
+        ImmutableMap.<String, String>builder()
+            .put(ID_KEY, StringIdentifier.fromId(uid).toString())
+            .put(BasePropertiesMetadata.GRAVITINO_MANAGED_ENTITY, Boolean.TRUE.toString())
+            .build();
+
+    SchemaEntity defaultSchema =
+        SchemaEntity.builder()
+            .withName(defaultSchemaIdent.name())
+            .withId(idGenerator.nextId())
```

Comment:
Why do you use another id, not the one generated above?

Reply:
fixed